### PR TITLE
Fix getting drop target when dragging from file manager

### LINKED ISSUE #28689: Drag and drop into subfolder uploads to current folder
### Steps

1. Create a folder "test"
1. Open the file manager (in my case Dolphin or Konqueror on Plasma 5)
1. Drag a file onto "test"

### Expected result
File is uploaded into "test"

### Actual result
File is uploaded inside the current directory.

### Version
ownCloud 10.0.3beta

Regression, this used to work correctly. Could be related to the move to new dav endpoint.

I tried the same with IE11 and the result is different but still bogus: https://github.com/owncloud/core/issues/28582

### PR DESCRIPTION
**Pull request by PVince81**
## Description
Possibly related to jquery.fileupload upgrade, now
e.originalEvent.target might be null.

## Related Issue
Fixes https://github.com/owncloud/core/issues/28689.

## Motivation and Context
See ticket

## How Has This Been Tested?
Manual testing like in the ticket.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: apps/files/js/filelist.js [hunk 1]
```diff
@@ -2708,7 +2708,7 @@
 					return false;
 				}
 
-				var dropTarget = $(e.originalEvent.target);
+				var dropTarget = $(e.originalEvent.target || e.originalEvent.delegatedEvent.target);
```
Comment (jvillafanez) [inline apps/files/js/filelist.js]: I guess the `e.original.target` won't be needed any longer.

The change in the library that affects this is https://github.com/owncloud/core/blob/639dafdd4db2eb72210b623b1a66c427b42b14d4/apps/files/js/jquery.fileupload.js#L1137 (before the library update) and https://github.com/owncloud/core/blob/master/apps/files/js/jquery.fileupload.js after the library update.
The actual event isn't set as the originalEvent but as the delegateEvent.

Unless we want compatibility between the library versions, which I don't think it's needed, we can remove the old code.

Comment (PVince81) [reply apps/files/js/filelist.js]: Thanks for looking deeper. To be honest I was just too lazy to investigate this detail.

Comment (PVince81) [reply apps/files/js/filelist.js]: @jvillafanez adjusted

**Comment by SergioBertolinSG** [comment]
When uploading several files to a folder there is no uploading bar.

**Comment by SergioBertolinSG** [comment]
When uploading a folder with several folders and files inside them, the files are not uploaded.

**Comment by PVince81** [comment]
stable10: https://github.com/owncloud/notifications/pull/117

**Comment by jvillafanez** [comment]
Wrong link? Backport doesn't seem related.

**Comment by PVince81** [comment]
oh lala, major browser tab fail. I'll fix

**Comment by jvillafanez** [comment]
> When uploading a folder with several folders and files inside them, the files are not uploaded.

Folder upload seems to be be supported only in chrome. It's very likely a browser limitation.

**Comment by SergioBertolinSG** [comment]
Using chrome I mean.

**Comment by PVince81** [comment]
- [ ] BUG: progress bar does not disappear when uploading a folder (chrome) into the current dir

**Comment by PVince81** [comment]
- [ ] BUG: target row size must be refreshed after transfer to have the correct size

I had a look at this and it's more complex, to be fixed separately. This is not related to the jquery.fileupload library update not recent changes. It is likely present since we moved to Webdav for file operations.

**Comment by PVince81** [comment]
@SergioBertolinSG please retest

**Comment by PVince81** [comment]
- [ ] fix js tests

**Comment by PVince81** [comment]
tests fixed

**Comment by SergioBertolinSG** [comment]
The estimation of the upload bar is very general (few seconds, minutes, etc) has it changed in this PR?

**Comment by SergioBertolinSG** [comment]
I guess it didn't so it works fine.

**Review by SergioBertolinSG — APPROVED** [review]
Works fine.

**Comment by PVince81** [comment]
@SergioBertolinSG the estimate was never good...

**Review by jvillafanez — APPROVED** [review]
Code looks good

**Comment by PVince81** [comment]
stable10: https://github.com/owncloud/core/pull/28882

**Pull request merged**